Source/dom/components/ImportableObject.js

```diff
@@ -0,0 +1,88 @@
+import { DefinedPropertiesKey, WrappedObject } from '../WrappedObject'
+import { Types } from '../enums'
+import { Factory } from '../Factory'
+import { wrapNativeObject } from '../wrapNativeObject'
+import { getDocumentData } from '../utils'
+
+const ObjectType = {
+  Symbol: 0,
+  LayerStyle: 1,
+  TextStyle: 2,
+  Unknown: 3,
+}
+
+export class ImportableObject extends WrappedObject {
+  /**
+   * Make a new symbol instance.
+   */
+  constructor(master = {}) {
+    if (!master.sketchObject) {
+      throw new Error(`Cannot create a ShareableObject directly`)
+    }
+    super(master)
+  }
+
+  importInDocument(document) {
+    const libraryController = AppController.sharedInstance().librariesController()
+
+    const documentData = getDocumentData(document)
+
+    const importedSymbol = libraryController
+      .importShareableObjectReference_intoDocument(this._object, documentData)
+      .localObject()
+
+    return wrapNativeObject(importedSymbol)
+  }
+
+  syncLocalDocumentInstancesWithLibrary(document) {
+    const libraryController = AppController.sharedInstance().librariesController()
+
+    const documentData = getDocumentData(document)
+
+    libraryController.syncForeignObject_withMaster_fromLibrary(
+      this._object.foreignObjectCollectionInDocument(documentData),
```

Comment:
I don't think this is right. `syncForeignObject:withMaster:fromLibrary:` takes a foreign object as it's first parameter but here we seem to be passing in the array of all foreign objects in the library.

If you really want to make `ImportableObject` sync-able you'd have to find the foreign object in the way that we do in `importShareableObjectReference:intoDocument:`. However, I'd suggest a better approach would be to add a `syncWithRemote` method to the symbol.

```
master = symbolRef.importInDocument(mydoc)
...
master.syncWithRemote()
```

The sync method on the master would be implemented something like (Sorry my JavaScript is poor, but hopefully you get the idea):

```
const libraryController = AppController.sharedInstance().librariesController()
const lib = libraryController.libraryForShareableObject( this._object )
const foreignObject = this._object.foreignObject
if (lib && foreignObject) {
  libraryController.syncForeignObject_withMaster_fromLibrary( foreignObject, nil, lib )
}
```

If we're add this to symbol we should handle the case where the symbol is local (`foreignObject` will be `nil`). Also, we can take advantage of the fact that if the second parameter is `nil` the code will find the remote from the library.

Reply:
> I'd suggest a better approach would be to add a syncWithRemote method to the symbol.

Yes good idea. the use case I had for this was to programmatically sync a library. So my thinking was to start from the library (> ImportableObject > sync). But I guess starting from the document works as well